Scripts/SAPSetupAssistantValidationScripts/SetupAssistantSAPValidateNetworkConnection.ps1

```diff
@@ -0,0 +1,29 @@
+param (
+   [string]$hostName,
+   [string]$port
+)
+
+$job = Start-Job -ScriptBlock {
+    param($hostName, $port)
+    Test-NetConnection -ComputerName $hostName -Port $port
+} -ArgumentList $hostName, $port
+
+for ($i = 0; $i -lt 5; $i++) {
+    Start-Sleep -Seconds 1
+}
+
+if (Wait-Job -Job $job -Timeout 5) {
+    $result = Receive-Job -Job $job
+
+    if ($result.TcpTestSucceeded) {
+        Write-Host -NoNewLine "[{'step': 'CheckNetworkConnection', 'status': 'Success', 'message': 'Successfully connected to %hostName% over port %port%.'}]"
```

Comment:
Command Prompt syntax usage detected.

PowerShell's string interpolation for variables should be as follows `$hostname` or IMHO `${}` when in a long string so there is clear delineation between content and variable.. Same with the port variable.

https://learn.microsoft.com/en-us/powershell/scripting/learn/deep-dives/everything-about-string-substitutions?view=powershell-7.4#delineation-with-braces